app/src/main/java/com/github/libretube/ui/views/SpinnerPreference.kt

```diff
@@ -0,0 +1,17 @@
+package com.github.libretube.ui.views
+import android.content.Context
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.preference.DropDownPreference
+import com.github.libretube.R
+
+class SpinnerPreference(context: Context, attrs: AttributeSet): DropDownPreference(context, attrs){
+
+    fun onCreate(parent: ViewGroup): View {
+        val inflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
+        return inflater.inflate(R.layout.spinner_preference, parent, false)
```

Comment:
Adding this preference in the settings doesn't seem to do anything. Also for some reason the onCreate function do not exist thus does not need to be overridden..

Reply:
Looks like we need to override `onCreateView` instead. And please inflate it like so:
```kotlin
val binding = SpinnerPreferenceBinding.inflate(LayoutInflater.from(context)
return binding.root
```

Reply:
`onCreateView` doesn't seem to exist either :/ Is this something unique to the preference classes? I would assume something like creating the view would be universal across UI elements

Reply:
I got it working like so:
```kotlin

package com.github.libretube.ui.views
import android.content.Context
import android.util.AttributeSet
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.preference.Preference
import com.github.libretube.R

class SpinnerPreference(context: Context, attrs: AttributeSet): Preference(context, attrs) {
    init {
        layoutResource = R.layout.spinner_preference
    }
}
```
So we need to subclass `Preference` now, otherwise it crashes in my tests.

The text has to be set in
```kotlin
override fun onBindViewHolder(holder: PreferenceViewHolder) {}
```
then.

Reply:
I have this so far (used chatgpt as a starter) Does this look correct?
`package com.github.libretube.ui.views

import android.content.Context
import android.content.res.TypedArray
import android.util.AttributeSet
import android.view.View
import android.widget.AdapterView
import androidx.preference.Preference
import androidx.preference.PreferenceViewHolder
import android.widget.ArrayAdapter
import android.widget.Spinner
import android.widget.TextView
import com.github.libretube.R

class SpinnerPreference(context: Context, attrs: AttributeSet) : Preference(context, attrs) {
    private lateinit var spinner: Spinner
    private lateinit var adapter: ArrayAdapter<CharSequence>
    private var selectedItem: CharSequence? = null

    init {
        layoutResource = R.layout.spinner_preference
    }

    override fun onBindViewHolder(holder: PreferenceViewHolder) {
        super.onBindViewHolder(holder)
        spinner = holder.itemView.findViewById(R.id.spinner)

        val titleView = holder.itemView.findViewById<TextView>(android.R.id.title) as? TextView
        titleView?.text = super.getTitle()

        val summaryView = holder.itemView.findViewById<TextView>(android.R.id.summary) as? TextView
        summaryView?.text = super.getSummary()

        // Set the spinner adapter
        adapter = ArrayAdapter.createFromResource(
            context,
            R.array.sb_skip_options,
            android.R.layout.simple_spinner_item
        )
        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
        spinner.adapter = adapter

        // Set the initial selected item
        if (selectedItem != null) {
            val position = adapter.getPosition(selectedItem!!)
            spinner.setSelection(position)
        }

        // Set a listener to handle item selection
        spinner.onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
            override fun onItemSelected(
                parent: AdapterView<*>,
                view: View?,
                position: Int,
                id: Long
            ) {
                selectedItem = adapter.getItem(position)
                persistString(selectedItem?.toString())
            }

            override fun onNothingSelected(parent: AdapterView<*>) {}
        }
    }

    override fun onGetDefaultValue(a: TypedArray, index: Int): Any {
        // Get the default value from the XML attribute, if specified
        return a.getString(index) ?: ""
    }

    override fun onSetInitialValue(defaultValue: Any?) {
        // Set the initial selected item from the persisted value, if available
        selectedItem = getPersistedString(defaultValue?.toString())
    }
}
`

Reply:
Seems fine